simulator/IsSimulatorPlugin.kt

```diff
@@ -0,0 +1,8 @@
+package venusbackend.simulator
+
+import venusbackend.riscv.MachineCode
+
+interface IsSimulatorPlugin {
+    /** called on every step of the simulation (after `sim.step()`) */
+    fun onStep(inst: MachineCode, prevPC: Number)
```

Comment:
why does this also not have the `finish` function call?

Reply:
because the return type is different

Reply:
But you can make the return type Any? so it can accept any return type.